src/components/PeopleFilters.tsx

```diff
@@ -1,96 +1,120 @@
+import { getSearchWith, SearchParams } from '../utils/searchHelper';
+import { CenturiesFilter } from '../types/CenturiesFilter';
+import classNames from 'classnames';
+import { SexFilter } from '../types/SexFilter';
+
 export const PeopleFilters = () => {
-  return (
-    <nav className="panel">
-      <p className="panel-heading">Filters</p>
+  const [searchParams, setSearchParams] = useSearchParams();
 
-      <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">
-          All
-        </a>
-        <a className="" href="#/people?sex=m">
-          Male
-        </a>
-        <a className="" href="#/people?sex=f">
-          Female
-        </a>
-      </p>
+  const setSearchWith = (paramsToUpdate: SearchParams) => {
+    const search = getSearchWith(searchParams, paramsToUpdate);
 
-      <div className="panel-block">
-        <p className="control has-icons-left">
-          <input
-            data-cy="NameFilter"
-            type="search"
-            className="input"
-            placeholder="Search"
-          />
+    setSearchParams(search);
+  };
 
-          <span className="icon is-left">
-            <i className="fas fa-search" aria-hidden="true" />
-          </span>
-        </p>
-      </div>
+  const sex = searchParams.get('sex');
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries');
 
-      <div className="panel-block">
-        <div className="level is-flex-grow-1 is-mobile" data-cy="CenturyFilter">
-          <div className="level-left">
-            <a
-              data-cy="century"
-              className="button mr-1"
-              href="#/people?centuries=16"
-            >
-              16
-            </a>
+  const queryHandler: ChangeEventHandler<HTMLInputElement> = e => {
+    setSearchWith({ query: e.target.value || null });
+  };
 
-            <a
-              data-cy="century"
-              className="button mr-1 is-info"
-              href="#/people?centuries=17"
-            >
-              17
-            </a>
+  return (
+    <div className="column is-7-tablet is-narrow-desktop">
+      <nav className="panel">
+        <p className="panel-heading">Filters</p>
 
-            <a
-              data-cy="century"
-              className="button mr-1 is-info"
-              href="#/people?centuries=18"
+        <p className="panel-tabs" data-cy="SexFilter">
+          {Object.entries(SexFilter).map(([key, value]) => (
+            <Link
+              key={value}
+              className={classNames({
+                'is-active': sex === value || (!sex && value === 'all'),
+              })}
+              to={{
+                search: getSearchWith(searchParams, {
+                  sex: value !== 'all' ? value : null,
+                }),
+              }}
             >
-              18
-            </a>
+              {key}
+            </Link>
+          ))}
```

Comment:
You're mapping over SexFilter and creating a Link for each entry. This part of the code could be extracted into a separate component, for example, a SexFilterLink component. This would improve readability and make your code more modular.